execution/aggregate/hashaggregate.go

```diff
@@ -225,7 +226,8 @@ func (a *aggregate) initializeScalarTables(ctx context.Context) ([]aggregateTabl
 
 	series = make([]labels.Labels, len(outputCache))
 	for i := 0; i < len(outputCache); i++ {
-		series[i] = outputCache[i].Metric
+		lbls, _ := function.DropMetricName(outputCache[i].Metric)
```

Comment:
I wonder why this wasn't a problem before.

Reply:
That is a really good question. Ill investigate a bit.

Reply:
With `by` we aggregate away the `__name__` label i guess

Reply:
Yeah i think its precisely that: https://github.com/prometheus/prometheus/blob/5ec1b4baaf01dae0b22567db53886c83b74e0e76/promql/engine.go#L2389

Reply:
`without` implicitely drops the MetricName label